tests/unit/test_algebraic_expression.cpp

```diff
@@ -174,30 +174,35 @@ class AlgebraicExpressionTest: public ::testing::Test {
 		mat_id = GraphContext_GetSchema(gc, "war", SCHEMA_EDGE)->id;
 		mat_ew = Graph_GetRelationMatrix(g, mat_id);
 
-        _matrices = raxNew();
-        raxInsert(_matrices, (unsigned char *)"p", strlen("p"), mat_p, NULL);
-        raxInsert(_matrices, (unsigned char *)"f", strlen("f"), mat_f, NULL);
-        raxInsert(_matrices, (unsigned char *)"c", strlen("c"), mat_c, NULL);
-        raxInsert(_matrices, (unsigned char *)"e", strlen("e"), mat_e, NULL);
-        raxInsert(_matrices, (unsigned char *)"F", strlen("F"), mat_ef, NULL);
-        raxInsert(_matrices, (unsigned char *)"V", strlen("V"), mat_ev, NULL);
-        raxInsert(_matrices, (unsigned char *)"W", strlen("W"), mat_ew, NULL);
+		GrB_Matrix dummy_matrix;
+		GrB_Matrix_new(&dummy_matrix, GrB_BOOL, 2, 2);
+		_matrices = raxNew();
+		raxInsert(_matrices, (unsigned char *)"p", strlen("p"), mat_p, NULL);
+		raxInsert(_matrices, (unsigned char *)"f", strlen("f"), mat_f, NULL);
+		raxInsert(_matrices, (unsigned char *)"c", strlen("c"), mat_c, NULL);
+		raxInsert(_matrices, (unsigned char *)"e", strlen("e"), mat_e, NULL);
+		raxInsert(_matrices, (unsigned char *)"F", strlen("F"), mat_ef, NULL);
+		raxInsert(_matrices, (unsigned char *)"V", strlen("V"), mat_ev, NULL);
+		raxInsert(_matrices, (unsigned char *)"W", strlen("W"), mat_ew, NULL);
+		raxInsert(_matrices, (unsigned char *)"A", strlen("A"), dummy_matrix, NULL);
+		raxInsert(_matrices, (unsigned char *)"B", strlen("B"), dummy_matrix, NULL);
+		raxInsert(_matrices, (unsigned char *)"C", strlen("C"), dummy_matrix, NULL);
 	}
 
 	AlgebraicExpression **build_algebraic_expression(const char *query) {
 		GraphContext *gc = QueryCtx_GetGraphCtx();
-        Graph *g = QueryCtx_GetGraph();
+		Graph *g = QueryCtx_GetGraph();
 		cypher_parse_result_t *parse_result = cypher_parse(query, NULL, NULL, CYPHER_PARSE_ONLY_STATEMENTS);
 		AST *master_ast = AST_Build(parse_result);
 		AST *ast = AST_NewSegment(master_ast, 0, cypher_ast_query_nclauses(master_ast->root));
 		QueryGraph *qg = BuildQueryGraph(gc, ast);
 		AlgebraicExpression **ae = AlgebraicExpression_FromQueryGraph(qg);
 
-        uint exp_count = array_len(ae);
-        for(uint i = 0; i < exp_count; i++) {
-            AlgebraicExpression_Optimize(ae + i);
-            _AlgebraicExpression_FetchOperands(ae[i], gc, g);
-        }
+		uint exp_count = array_len(ae);
+		for(uint i = 0; i < exp_count; i++) {
+			_AlgebraicExpression_FetchOperands(ae[i], gc, g);
+			AlgebraicExpression_Optimize(ae + i);
```

Comment:
`AlgebraicExpression_Optimize` calls `_AlgebraicExpression_FetchOperands`
and so the call to `_AlgebraicExpression_FetchOperands` should be dropped.